enhancements/update/upgrades-blocking-on-ack.md

```diff
@@ -0,0 +1,162 @@
+
+* As an OCP developer I want a way to introduce additional "acknowledgment based" upgrade blocks for blocking minor level upgrades.
+
+* As an OCP product team we do not want customers to be surprised by workloads breaking due to APIs being removed after they upgrade.
+
+### Implementation Details/Notes/Constraints
+
+### Risks and Mitigations
+
+* A common risk is human error resulting in an incorrect key/value pair being placed in the configmap.
+This type of error will not change `Upgradeable=false` and therefore the administrator will know a mistake has been made since the `Upgradeable` condition message will continue to be displayed.
+CVO will log an error whenever an unrecognized key is found in the configmap.
+* Another risk is that the configmap gets removed.
+This will result in `Upgradeable=false` and the standard `Upgradeable` condition message will be displayed with the additional message that the configmap cannot be found and should be created.
+CVO will log an error whenever the configmap has been removed.
+* Unable to access the configmap.
+This will result in `Upgradeable=false` and the standard `Upgradeable` condition message will be displayed with the additional message that the configmap cannot be accessed.
+CVO will log an error whenever there is an issue accessing the configmap.
+
+## Design Details
+
+### Test Plan
+
+* CVO unit tests will be expanded and/or created to test the new logic.
+* e2e upgrade test will be enhanced.
+
+### Graduation Criteria
+
+GA. When it works, we ship it.
+
+#### Dev Preview -> Tech Preview
+
+N/A. This is not expected to be released as Dev Preview.
+
+#### Tech Preview -> GA
+
+N/A. This is not expected to be released as Tech Preview.
+
+#### Removing a deprecated feature
+
+### Upgrade / Downgrade Strategy
+
+Inherent in the key name is an OCP major/minor version number. Based on this version number CVO knows to which OCP versions a given key applies.
+
+### Version Skew Strategy
+
+No special consideration.
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation
+History`.
+
+## Drawbacks
+
+* CVO will contain a hard-coded key name for each specific acknowledge based upgrade block.
+Configuring these keys through one or more configuration configmaps was considered but decided against due to the added complexity of maintaining and consuming the additional configmaps.
+* Keys will not be portable across products that use a different versioning scheme than OCP.
+* `Upgradeable=false` is the default state and results if there is any issue accessing the configmap or a key is invalid.
+
+## Alternatives
+
+[k8s-1.22-removed-apis]: https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-22
+[upgradeableCheck-interface]: https://github.com/jottofar/cluster-version-operator/blob/e1adca47977bbea635b7fb2dc2473525e512a539/pkg/cvo/upgradeable.go#L132
```

Comment:
not clear why this is under "alternatives"

Reply:
It's not. It's just the link definitions. They only show up when file is viewed in raw form. I'll pust something under the `Alternatives` heading so it is not confusing in the raw.